docs/testing-strategy.md

```diff
@@ -0,0 +1,154 @@
+# Testing Strategy
+
+We aim for Content Publisher to have a test suite that provides developers
+with value. We want the application to be fast to test; for the tests to catch
+breaking behavioural changes; and for it to be easy for developers to know
+where to add tests. The purpose of this document is
+to provide a summary of the testing approaches used so that new and old
+developers to Content Publisher have a shared understanding.
+
+The principles we most value in testing are:
+
+- consistency across tests over developer preference - all tests should
+  feel like they could have been written by the same person;
+- readable tests that are easy to follow - aim for expectations to be clear,
+  setup code to be close to associated assertions and avoid being overzealous
+  in removing repetition;
+- avoid testing configuration - tests should be focused on testing logic we've
+  introduced, rather than whether or not our code was typed correctly;
+- pragmatism in what we test - if all code paths of a class/module are
```

Comment:
I agree with this, but I think it would be helpful if existing tests were sign posted somewhere, like the commit message?

Reply:
I wouldn't say that this is a concern as part of test strategy (as I don't think we've pushed people to do so) so I don't think it belongs here. It sounds like a good courtesy though and be interesting to keep an eye out for if we think we have a problem in this regard.